### DISCUSSION TITLE
Fault tolerance/resilience study of Fprime

### DISCUSSION THREAD

**Discussion by hjiang13** [nasa/fprime#791] [Q&A]
Our team is working on the resilience study of applications/systems.
We learned that F prime has a consideration of fault tolerance for their applications.
What we are curious about is there any design or module for fault tolerance in Fprime framework? 
 Also, Fprime is applied to multiple projects. Can you provide a few projects for us to study his resilience characteristics? Thank you
Any response would be appreciated.

**Comment by LeStarch** [comment]
Each project typically architects its own fault tolerance approach as each project's requirements are different.  We provide a few basic fault handling components https://github.com/nasa/fprime/tree/devel/Svc/AssertFatalAdapter and https://github.com/nasa/fprime/tree/devel/Svc/FatalHandler to handle fatal system events, but this is typically not sufficient for any moderately complex project.

I don't know of any projects to suggest looking at, but others can comment here if they have a project to suggest.
